compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/Utils.java

```diff
@@ -251,4 +261,31 @@ public static boolean isContextParameter(TypeSymbol typeSymbol) {
     public static String getAccessor(ResourceMethodSymbol resourceMethodSymbol) {
         return resourceMethodSymbol.getName().orElse(null);
     }
+
+    public static boolean isGraphQLServiceObjectDeclaration(
+            ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        TypedBindingPatternNode typedBindingPatternNode = moduleVariableDeclarationNode.typedBindingPattern();
+        TypeDescriptorNode typeDescriptorNode = typedBindingPatternNode.typeDescriptor();
+        if (typeDescriptorNode.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+            return false;
+        }
+        return isGraphqlServiceQualifiedNameReference((QualifiedNameReferenceNode) typeDescriptorNode);
+    }
+
+    private static boolean isGraphqlServiceQualifiedNameReference(QualifiedNameReferenceNode nameReferenceNode) {
+        Token modulePrefixToken = nameReferenceNode.modulePrefix();
+        if (modulePrefixToken.kind() != SyntaxKind.IDENTIFIER_TOKEN) {
+            return false;
+        }
+        if (!PACKAGE_NAME.equals(modulePrefixToken.text())) {
+            return false;
+        }
+        IdentifierToken identifier = nameReferenceNode.identifier();
+        return SERVICE_NAME.equals(identifier.text());
+    }
+
+    public static boolean isFunctionDefinition(Node node) {
+        return node.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION
+                || node.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION;
```

Comment:
What's the purpose of this method? The name and the logic don't seem to be matching.

Reply:
Purpose of this method is to check provided node is a FunctionDefinitionNode in an object

Reply:
But this returns whether it is a resource accessor definition or an object method definition?

Reply:
resource accessor definition and object method definition are the only 2 method definition we could have in an object right? both are instances of FunctionDefinitionNode.